src/plugins/data/public/antlr/shared/utils.ts

```diff
@@ -93,41 +96,82 @@ export const fetchData = (
   });
 };
 
-// Specific fetch function for table schemas
-// TODO: remove this after using data set table schema fetcher
-export const fetchTableSchemas = (tables: string[], api: any, queryString: QueryStringContract) => {
-  return fetchData(
-    tables,
-    (table, dataSourceId, title) => ({
-      query: { query: `DESCRIBE TABLES LIKE ${table}`, format: 'jdbc' },
-      df: {
-        meta: {
-          queryConfig: {
-            dataSourceId: dataSourceId || undefined,
-            title: title || undefined,
-          },
+export const fetchColumnValues = async (
+  table: string,
+  column: string,
+  services: IDataPluginServices,
+  fieldInOsd: IndexPatternField | undefined
+): Promise<any[]> => {
+  // default to true/false values for type boolean
+  if (fieldInOsd?.type === 'boolean') {
+    return ['true', 'false'];
+  }
+
+  const allowedType = ['string'];
+  // don't return values if ui settings prevent it or the field type isn't allowed
+  if (
+    !services.uiSettings.get(UI_SETTINGS.QUERY_ENHANCEMENTS_SUGGEST_VALUES) ||
+    !fieldInOsd ||
+    !allowedType.includes(fieldInOsd.type)
+  ) {
+    return [];
+  }
+  const limit = services.uiSettings.get(UI_SETTINGS.QUERY_ENHANCEMENTS_SUGGEST_VALUES_LIMIT);
+
+  // get dataset for connecting to the cluster currently engaged
+  const dataset = services.data.query.queryString.getQuery().dataset;
+
+  return (
+    await fetchFromAPI(
+      services.http,
+      JSON.stringify({
+        query: {
+          query: `SELECT ${column} FROM ${table} GROUP BY ${column} ORDER BY COUNT(${column}) DESC LIMIT ${limit}`,
+          language: 'SQL',
+          format: 'jdbc',
+          dataset,
         },
-      },
-    }),
-    api,
-    queryString
-  );
+      })
+    )
+  ).body.fields[0].values;
```

Comment:
In the case when `fetchFromAPI` rejects the promise with the error, the rejected promise will travel up to the caller of `fetchColumnValues`. I wrote a test for this mocking an error from the http fetch function that `fetchFromAPI` calls:
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8275/files/2a167e91cc27b7be0091cd010495ed87ef9cd0e7#diff-b547688af7b19722a725a97405371301f010a16e1d6dd8dfbfdae74b0e3c784eR374-R382 

In the case where the fetched object doesn't match this caller chain, it will throw a rejected promise about the format. I wrote it this way because currently the returned object has this exact format, and there is no other standard to flatted this object to that I'm aware of.